### ISSUE TITLE
Configuration for private accounts in separate configuration

### ISSUE THREAD

**Issue by sideninja** [onflow/flow-cli#70]
### Issue To Be Solved

We believe the CLI configuration file `flow.json` should be included in the versioning control (github...) similar to how for example `package.json` is.

Since configuration can include sensitive private keys we should allow developers to separate the configuration for their testnet and mainnet accounts from the main `flow.json` configuration so they are not mistakenly committed to a versioning control systems. 

Currently, the configuration looks like this (where secret gets constantly replaced before committing - a ticking bomb):

```
...
	"accounts": {
		"testnet-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "SECRET"
		}
	},
...
```

### Suggest A Solution

After a brief discussion with @psiemens we believe there are at least the following solutions to this problems:

#### Seperate Private Config File
We can create a separate configuration file - let's call it `flow.private.json` for the sake of the example and this file could include any number of accounts that would on the runtime of CLI command be added to the original `flow.json` configuration (but not saved to the file). Use of this would happen "in-memory". 

Example for `flow.private.json`:

```
...
	"accounts": {
		"testnet-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "ACTUAL_SECRET_WOULD_BE_HERE"
		}
	},
...
```

And then `flow.json` could look like:

```
...
	"accounts": {
                "emulator-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "11c5dfdeb0ff03a7a73ef39788563b62c89adea67bbb21ab95e5f710bd1d40b7",
			"chain": "flow-emulator"
		},
		"testnet-account": {
                       "private": true
                }
	},
...
```

Or another option:

```
...
	"accounts": {
                "emulator-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "11c5dfdeb0ff03a7a73ef39788563b62c89adea67bbb21ab95e5f710bd1d40b7",
			"chain": "flow-emulator"
		},
		"testnet-account": "flow.private.json"
	},
...
```
There is also a possibility to not change anything in flow.json and just for the CLI to figure out if the file flow.private.json exists and overwrite it on the fly.

#### Environment variable support

We could also allow passing account secret as an environment variable. This can be a standalone solution or it can work nicely with the above solution.

Example:

`flow project deploy --acount=testnet --secret=123`

Example `flow.json`:

```
...
	"accounts": {
		"testnet-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "$ENV.secret"
		}
	},
...
```

**Comment by bjartek** [comment]
I like the idea of beeing able to specify a "private" json file as an cli argument and just merge it in. Or using env vars, but work really well.

**Comment by psiemens** [comment]
@sideninja I've thought about this a bit more. I think we should implement the `flow.private.json` solution in an upcoming sprint because it's better UX IMO, and also will allow for more flexibility when defining account keys. 

However, I think the highest priority is to give users a way to securely specify a few pieces of private data, and that can be solved easily and quickly with environment variable support. Do you think we could implement support for environment variables in this sprint, and then do `flow.private.json` as a follow-up feature?

I was looking at other examples and it looks like this syntax is quite common:

```json
...
	"accounts": {
		"testnet-account": {
			"address": "f8d6e0586b0a20c7",
			"keys": "${env:MY_PRIVATE_KEY}"
		}
	},
...
```

A few examples:
- https://code.visualstudio.com/docs/editor/variables-reference#_environment-variables
- https://www.serverless.com/framework/docs/providers/aws/guide/variables#referencing-environment-variables
- https://www.elastic.co/guide/en/beats/winlogbeat/current/using-environ-vars.html

**Comment by bjartek** [comment]
Another option is to just use a convenience here FLOW_TESTNET_<NAME>_KEY aso

**Comment by sideninja** [comment]
> Another option is to just use a convenience here FLOW_PK_TESTNET-ACCOUNT aso

What do you mean by that?

**Comment by bjartek** [comment]
I. The given example you have sn account test-account that need a PK. If it is not present look up an env var based on convention and use that instead.

One convention could be FLOW_ACCOUNT_PK as a prefix and the the mame of the account after that.

However the env lookup works as well.

**Comment by sideninja** [comment]
I see. The way I actually implemented currently is to have an option to specify any environment variable and include in the config. 

Example:
`EMULATOR_PK=123 flow project deploy` 

Config:
```
...
"service-account": {
  "address": "f8d6e0586b0a20c7",
  "keys": "${env:EMULATOR_PK}"
}
...
```

This way we can actually define and use any env variable in config. Even in other places like so:

Example:
`EMULATOR_ADDRESS=0x123 flow project deploy` 

Config:
```
...
"service-account": {
  "address": "${env:EMULATOR_ADDRESS}",
  "keys": "123123"
}
...
```

I think this might be better than having any special meaning env variables since it will allow general use and no further instructions. 

I also used standard syntax for passing env variables and not `--secret=123` because I think it is better we make a clear difference between flags and env variables. Flags should be imho used as a predefined set of values that are included in help and have predefined values. Whereas env variables are normally used the way I've implemented them and thus allowing better support for different environments where you set env variables through an interface (gcloud etc). I think it is safer and better to have them separate.
